proposed/container.md

```diff
@@ -39,21 +39,14 @@ An entry identifier is any PHP-legal string of at least one character that uniqu
 - `has` takes one unique parameter: an entry identifier, which MUST be a string.
   `has` MUST return `true` if an entry identifier is known to the container and `false` if it is not.
   `has($id)` returning true does not mean that `get($id)` will not throw an exception.
-  It does however mean that `get($id)` will not throw a `NotFoundExceptionInterface`.
 
 ### 1.2 Exceptions
 
 Exceptions directly thrown by the container SHOULD implement the
 [`Psr\Container\ContainerExceptionInterface`](#container-exception).
 
-A call to the `get` method with a non-existing id MUST throw a
-[`Psr\Container\NotFoundExceptionInterface`](#not-found-exception).
-
-A call to `get` can trigger additional calls to `get` (to fetch the dependencies).
-If one of those dependencies is missing, the `NotFoundExceptionInterface` triggered by the
-inner `get` call SHOULD NOT bubble out. Instead, it should be wrapped in an exception 
-implementing the `ContainerExceptionInterface` that does not implement the 
-`NotFoundExceptionInterface`.
+A call to the `get($id)` MUST throw a [`Psr\Container\NotFoundExceptionInterface`](#not-found-exception) if and only 
```

Comment:
Actually, reading back this sentence, @schnittstabil is right.

I wrote:

> A call to the `get($id)` MUST throw a [`Psr\Container\NotFoundExceptionInterface`](#not-found-exception) if and only if `has($id)` returns `false`.

So what happens if `has` returns `true`? The opposite of MUST is CAN.

It means that:

> `get($id)` CAN throw a `Psr\Container\NotFoundExceptionInterface` if `has` returns `false`

So to me, the correct wording for this is:

> A call to the `get($id)` throws a [`Psr\Container\NotFoundExceptionInterface`](#not-found-exception) if and only if `has($id)` returns `false`.

In general, when using "if and only if", the part to the left and the part to the right must be "facts".